backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/computing/purchasecontract/interim/InterimComputingMethod.java

```diff
@@ -169,10 +169,9 @@ public boolean applies(final @NonNull TableRecordReference recordRef, @NonNull f
 		final Quantity qty = Quantity.of(BigDecimal.ZERO, stockUOM);
 		final List<ModularContractLogEntry> logs = computingMethodService.retrieveLogsForCalculation(request);
 
-		computingMethodService.validateLogs(logs);
-		logs.forEach((log) -> qty.add(computingMethodService.getQtyToAdd(log, request.getProductId())));
+		final ProductPrice logProductPrice = computingMethodService.getUniqueProductPriceOrError(logs).orElse(null);
 
-		final ProductPrice logProductPrice = logs.get(0) != null ? logs.get(0).getPriceActual() : null;
+		logs.forEach((log) -> qty.add(computingMethodService.getQtyToAdd(log, request.getProductId())));
```

Comment:
Quantity is immutable so qty.add will return a new instance.

not sure what are u trying to achieve here, but might be that `reduce` is what u want to use instead of `forEach`

Reply:
I've just deleted a line here :)